### PR TITLE
Update the grouping logic for categories

### LINKED ISSUE #12410: Update how ShortcodeDescriptors generate a list of categories
### Describe the bug
In the [Shortcode views,](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/ShortcodeModal.cshtml#L13) update the code that groups the categories.

```
var categories = shortcodeDescriptors
        .Where(s => s != null) // this check is for each shortcodeDescriptor and should be for categories
        .SelectMany(s => s.Categories)
        .OrderBy(c => c, StringComparer.OrdinalIgnoreCase)
        .GroupBy(c => c, StringComparer.OrdinalIgnoreCase)
        .Select(g => g.Key);
```

The code can be simplify to this

```
var categories = shortcodeDescriptors.SelectMany(x => x.Categories ?? Enumerable.Empty<string>())
    .Distinct(StringComparer.OrdinalIgnoreCase)
    .OrderBy(category => category);
```

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #12410

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/ShortcodeModal.cshtml [hunk 1]
```diff
@@ -81,7 +80,7 @@ and call @await DisplayAsync(await New.ShortcodeModal()) from the fields view
             </div>
         </div>
     </zone>
-   <script depends-on="shortcodes" at="Foot">
+    <script depends-on="shortcodes" at="Foot">
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/ShortcodeModal.cshtml:83]: Extra space

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/ShortcodeModal.cshtml [hunk 2]
```diff
@@ -10,25 +10,24 @@ and call @await DisplayAsync(await New.ShortcodeModal()) from the fields view
...
     var shortcodeDescriptors = await ShortcodeDescriptorManager.GetShortcodeDescriptors();
-    var categories = shortcodeDescriptors
-        .Where(s => s != null)
-        .SelectMany(s => s.Categories)
-        .OrderBy(c => c, StringComparer.OrdinalIgnoreCase)
-        .GroupBy(c => c, StringComparer.OrdinalIgnoreCase)
-        .Select(g => g.Key);
+
+    var categories = shortcodeDescriptors.SelectMany(shortcodeDescriptor => shortcodeDescriptor.Categories ?? Enumerable.Empty<string>())
+    .Distinct(StringComparer.OrdinalIgnoreCase)
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/ShortcodeModal.cshtml:15]: Intend lines 15-17

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Edit.cshtml [hunk 3]
```diff
@@ -60,55 +59,55 @@
...
-                    DefaultValue = Model.DefaultValue,
-                    Usage = Model.Usage           
-                };
-                dynamic shape = await ShortcodeDescriptorDisplayManager.BuildDisplayAsync(descriptor, null, "SummaryAdmin");
-                shape.Actions = null;
-                @await DisplayAsync(shape)
-            }
+                @{
+                    var descriptor = new ShortcodeDescriptor
+            {
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Edit.cshtml:101]: Intend the initializer block

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Edit.cshtml [hunk 4]
```diff
@@ -60,55 +59,55 @@
...
-                        <template slot="noResult">
-                            @T["No categories found"]
-                        </template>
-                        <template slot="noOptions">
-                            @T["No categories found"]
-                        </template>
-                    </vue-multiselect>
+                <input class="form-control" asp-for="SelectedCategories" type="hidden" v-bind:value="getSelectedCategories()" />
+                <vue-multiselect v-model="value"
+                                 placeholder="@T["Type to search"]"
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Edit.cshtml:69]: Remove extra whitespace

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Edit.cshtml [hunk 5]
```diff
@@ -4,13 +4,12 @@
 
 @{
     var shortcodeDescriptors = await ShortcodeDescriptorManager.GetShortcodeDescriptors();
-    var categories = shortcodeDescriptors
-        .Where(s => s != null)
-        .SelectMany(s => s.Categories)
-        .GroupBy(c => c, StringComparer.OrdinalIgnoreCase)
-        .Select(g => g.Key);
+    var categories = shortcodeDescriptors.SelectMany(shortcodeDescriptor => shortcodeDescriptor.Categories ?? Enumerable.Empty<string>())
+    .Distinct(StringComparer.OrdinalIgnoreCase)
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Edit.cshtml:8]: Intend lines 8-10

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml [hunk 6]
```diff
@@ -57,49 +57,49 @@
...
-                        <template slot="noResult">
-                            @T["No categories found"]
-                        </template>
-                        <template slot="noOptions">
-                            @T["No categories found"]
-                        </template>
-                    </vue-multiselect>
+                <input class="form-control" asp-for="SelectedCategories" type="hidden" v-bind:value="getSelectedCategories()" />
+                <vue-multiselect v-model="value"
+                                 placeholder="@T["Type to search"]"
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml:67]: Remove extra spaces

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
     var shortcodeDescriptors = await ShortcodeDescriptorManager.GetShortcodeDescriptors();
-    var categories = shortcodeDescriptors
-        .Where(s => s != null)
-        .SelectMany(s => s.Categories)
-        .GroupBy(c => c, StringComparer.OrdinalIgnoreCase)
-        .Select(g => g.Key);
 
-    var jCategories = JsonConvert.SerializeObject(categories);  
+    var categories = shortcodeDescriptors.SelectMany(shortcodeDescriptor => shortcodeDescriptor.Categories ?? Enumerable.Empty<string>())
+    .Distinct(StringComparer.OrdinalIgnoreCase)
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml:8]: Intend lines 8-10

**Review by hishamco — CHANGES_REQUESTED** [review]

**Comment by MikeAlhayek** [comment]
@hishamco I undid the changes that were auto done by Visual Studio.

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml [hunk 8]
```diff
@@ -1,15 +1,15 @@
 @model ShortcodeTemplateViewModel
 @inject IDisplayManager<ShortcodeDescriptor> ShortcodeDescriptorDisplayManager
 @inject IShortcodeDescriptorManager ShortcodeDescriptorManager
-@{
+@{
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml:4]: Extra space

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml [hunk 9]
```diff
@@ -57,49 +57,49 @@
                 <label asp-for="Usage">@T["Usage"]</label>
                 <textarea asp-for="Usage" rows="10" class="form-control"></textarea>
                 <span class="hint">@T["Example usage of the shortcode."]</span>
-            </div>
+            </div>
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml:60]: Extra space

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml [hunk 10]
```diff
@@ -57,49 +57,49 @@
...
                 <span class="hint">@T["The categories for this shortcode."]</span>
-            </div>
+            </div>      
 
             <div class="mb-3" asp-validation-class-for="DefaultValue">
                 <label asp-for="DefaultValue">@T["Default Shortcode Value"]</label>
                 <input asp-for="DefaultValue" class="form-control" />
                 <span class="hint">@T["The default value of the shortcode when selected. Defaults to [name]."]</span>
-            </div>
+            </div>
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml:93]: Extra space

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml [hunk 11]
```diff
@@ -129,7 +129,7 @@
 
 <script at="Foot">
     initializeShortcodesTemplateEditor(
-        document.getElementById('shortcodeCategories'),
+        document.getElementById('shortcodeCategories'),
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml:132]: Extra space

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Edit.cshtml [hunk 12]
```diff
@@ -4,12 +4,13 @@
...
-    .OrderBy(category => category)
-    .ToList();
+    var categories = shortcodeDescriptors
+        .SelectMany(shortcodeDescriptor => shortcodeDescriptor.Categories ?? Enumerable.Empty<string>())
+        .Distinct(StringComparer.OrdinalIgnoreCase)
+        .OrderBy(category => category)
+        .ToList();
 
-    var jCategories = JsonConvert.SerializeObject(categories);
+    var jCategories = JsonConvert.SerializeObject(categories);
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Edit.cshtml:13]: Extra space

**Review by hishamco — CHANGES_REQUESTED** [review]
Please fix the format everywhere

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml (hunk 8 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Shortcodes/Views/Admin/Create.cshtml:4]: I don't see these spaces in the diff.

**Pull request merged**